### PR TITLE
W-10423588 - Remove Soft Credits

### PR DESCRIPTION
**Pull request by mcasco**
# Critical Changes
    Refactor to delete Soft and Partial soft credit builders.
# Changes
    Delete all method for Soft and PartialSoft credits.

### CODE REVIEW

**Comment by mcasco** [comment]
[W-10423588](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000egS7LYAU/view)

**Review by JoseRossiSF — APPROVED** [review]
LGTM

File: force-app/main/service/DonationHistoryService.cls [hunk 1]
```diff
@@ -35,16 +35,12 @@
 public with sharing class DonationHistoryService {
     
     static final String NPSP_CREDIT_ROLES = 'NPSP_OppsSoftCreditRoles';
```
Comment (bdvorachek) [inline force-app/main/service/DonationHistoryService.cls:37]: This constant is unused and we should remove it.

File: force-app/main/service/DonationHistoryService.cls [hunk 2]
```diff
@@ -35,16 +35,12 @@
...
-    private List<OpportunityContactRole> hardCredits;
-    private List<OpportunityContactRole> softCredits;
-    private List<OpportunityContactRole> partialSoftCredits;
+    private List<OpportunityContactRole> credits;
```
Comment (bdvorachek) [inline force-app/main/service/DonationHistoryService.cls:38]: This variable is always an empty list and can be removed.

**Review by bdvorachek — CHANGES_REQUESTED** [review]

File: force-app/main/service/DonationHistoryService.cls (hunk 1 above)
Comment (mcasco) [reply force-app/main/service/DonationHistoryService.cls:37]: constant removed!

File: force-app/main/service/DonationHistoryService.cls (hunk 2 above)
Comment (mcasco) [reply force-app/main/service/DonationHistoryService.cls]: variable removed!

**Review by dcaballeroSf — APPROVED** [review]
LGTM

File: force-app/main/service/DonationHistoryService.cls [hunk 3]
```diff
@@ -33,18 +33,12 @@
...
 */
 public with sharing class DonationHistoryService {
-    
-    static final String NPSP_CREDIT_ROLES = 'NPSP_OppsSoftCreditRoles';
-    private List<OpportunityContactRole> hardCredits;
-    private List<OpportunityContactRole> softCredits;
-    private List<OpportunityContactRole> partialSoftCredits;
-    private Id contactId;
+
+    private List<Opportunity> credits;
```
Comment (bdvorachek) [inline force-app/main/service/DonationHistoryService.cls:37]:
```suggestion
    private List<Opportunity> credits;
    private Id contactId;
```

**Review by bdvorachek — CHANGES_REQUESTED** [review]
There's a build error here that we need to fix, but otherwise I think this is ok!

File: force-app/main/service/DonationHistoryService.cls (hunk 3 above)
Comment (mcasco) [reply force-app/main/service/DonationHistoryService.cls]: contactId added.

**Review by bdvorachek — APPROVED** [review]

**Review by psadie-sfdo — APPROVED** [review]

**Pull request merged**